### PR TITLE
Check that field type and it's default value have matching types.

### LINKED ISSUE #1598: Type annotation in model is ignored when there is a default value
In this model, the field `isCool` is inferred to have type `number` (from the default value), even though the explicit type annotation says `boolean`:

```ts
import { ChiselEntity } from '@chiselstrike/api';

export class Human extends ChiselEntity {
    name: string;
    isCool: boolean = 1;
}
```

Expected behavior is either that `isCool` is a `boolean` or an error. Note that typechecking fails in this case.

### PR DESCRIPTION
**Pull request by BearLemma**
Closes #1598.

### CODE REVIEW

File: cli/tests/integration_tests/rust_tests/model_defaults.rs [hunk 1]
```diff
@@ -260,3 +260,65 @@ pub async fn complex_optional_defaults(c: TestContext) {
...
+    );
+    c.chisel.apply_err().await;
+
+    c.chisel.write(
+        "models/default.ts",
+        r##"
+        import { ChiselEntity } from "@chiselstrike/api";
+
+        export class Defaulted extends ChiselEntity {
+            a: bool = "true";
```
Comment (honzasp) [inline cli/tests/integration_tests/rust_tests/model_defaults.rs:309]: The correct syntax is `boolean`, so this file produces an error, but it is probably different from the error that we want to test for. Perhaps we should check the error returned from `apply_err().await`, to make the tests stronger?

File: cli/tests/integration_tests/rust_tests/model_defaults.rs [hunk 2]
```diff
@@ -260,3 +260,65 @@ pub async fn complex_optional_defaults(c: TestContext) {
     )
     .unwrap();
 }
+
+#[chisel_macros::test(modules = Deno)]
+pub async fn field_type_and_value_type_mismatch(c: TestContext) {
```
Comment (honzasp) [inline cli/tests/integration_tests/rust_tests/model_defaults.rs:265]: What happens when I use a more complex expression, for example `x: string = 1 + 1` or `x: number = function() {}`? Do we test this somewhere else? If not, perhaps we should add these tests here.

Comment (honzasp) [reply cli/tests/integration_tests/rust_tests/model_defaults.rs:265]: Ok, what about the `Expr::Unary` case? If I look at the code, it seems that we would accept `x: number = !42` or `x: boolean = typeof true`?

Comment (honzasp) [reply cli/tests/integration_tests/rust_tests/model_defaults.rs:265]: Are there some tests for that case?

**Review by honzasp — APPROVED** [review]

File: cli/tests/integration_tests/rust_tests/model_defaults.rs (hunk 2 above)
Comment (dekimir) [reply cli/tests/integration_tests/rust_tests/model_defaults.rs:265]: > Ok, what about the `Expr::Unary` case? If I look at the code, it seems that we would accept `x: number = !42` or `x: boolean = typeof true`?

#1291

Comment (dekimir) [reply cli/tests/integration_tests/rust_tests/model_defaults.rs:265]: Ideally, the parser would recognize a negative number for what it is: a literal, not an expression. IDK if that's a viable option.

**Pull request merged**